### PR TITLE
Treeview: Fix highlighting of active icon during navigatoins

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #12916

This PR addresses issues where the active item in the tree view was not updated when navigating. We do this by specifically:

- When navigating to a parent directory, we now wait for the content to load and try to select the item in the loaded list (this won't work if we navigate to the parent directory because the root tree entry we're looking at is not rendered within the tree (e.g. if you navigate to `/foo/bar` we don't have an item for `/foo/bar` in the list. This is fine though!).
- When navigating to a non-loaded file within the current tree, we now load the path to that entry with all it's ancestors. This allows us to connect it to the current view even if there are directories in between. We already skip items that have been added already so this seems to work alright (will upload a test video later). 
- Another fix here is that items that are nested within collapsed folders will now expand when you navigate to the.

This also fixes an issue in the `react-accessible-treeview `component where it was causing selected item to always gain focus. The component should only handle focus if it already has focus somewhere inside the tree. This is a change that is also necessary for searching symbols.
 
## Test plan

https://user-images.githubusercontent.com/458591/219373228-3e97239a-61e3-429a-a522-f8c9ef219b62.mov

Lots of manuel testing but this is still behind a feature flag that is only enabled on dotcom and S2 so we'll dogfood.

## App preview:

- [Web](https://sg-web-ps-tree-view-navigation.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.15 kb) | 0.00% (+0.15 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/tree-view-navigation/compare-report.html;tab=live_object) for a full comparison between the commits 453e789 and 7f875f1 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/RepoRevisionSidebarFileTree.tsx [hunk 1]
```diff
@@ -215,28 +228,92 @@ export const RepoRevisionSidebarFileTree: React.FunctionComponent<Props> = props
...
+        const treeData = treeDataRef.current
+        if (!treeData) {
+            return
+        }
+
+        function selectAndExpandPathToNode(path: string): boolean {
+            const treeData = treeDataRef.current
+            if (!treeData) {
+                return false
+            }
```
Comment (taras-yemets) [inline client/web/src/repo/RepoRevisionSidebarFileTree.tsx:244]: We can probably remove this check or the one above (235-238).

**Review by taras-yemets — APPROVED** [review]
Nice!
What do you think about scrolling the selected item into view (28-35 sec on demo video)?

**Comment by philipp-spiess** [comment]
> What do you think about scrolling the selected item into view (28-35 sec on demo video)?

It's actually quite concerning that this does not work yet because _it should_. Oh no :( I'll look into it as a follow-up

**Pull request merged**